rust/graph-triples/src/resources.rs

```diff
@@ -88,15 +88,59 @@ impl Resource {
     }
 }
 
+/// Under which circumstances a [`Resource`] should be automatically run.
+///
+/// In the below descriptions:
+///
+/// - "run" means that the user made an explicit request to execute the specific resource
+///   (e.g. presses the run button on a `CodeChunk`), or the containing resource (e.g. presses
+///   the run button on the parent `Article`).
+///
+/// - "autorun" means that the resource is automatically executed, without an explicit
+///   user request do so (but in some cases in response to one).
+#[derive(Debug, Clone, Serialize)]
+pub enum ResourceAutorun {
+    /// Never automatically execute the resource.
+    /// Only execute when the user explicitly runs the resource (or its containing resource).
+    ///
+    /// e.g. a user may tag a `CodeBlock` as `@autorun never` if it is long running
+    /// and they want to check the outputs of previous code chunks before proceeding
```

Comment:
Just to confirm, this means that if `CodeChunk a` is marked as `@autorun never`, but is referenced from `CodeChunk b`. Changes to `a` mark both `a` and `b` as stale, and no matter how many times you run `b` it will always be in a stale status.
User would have to manually run `a`. Then `b` would react to the the update and get automatically re-evaluated?

We could explore showing a warning or something for such cases, and offer to re-run stale upstream nodes for the user.

Reply:
Yes, your reiteration is correct. It is also a good example of why it is useful to have separate `executeRequired` and `exexuteStatus` properties. The user might run b to check that it is correct, ie has status `Succeeded` but still want to know that `executeRequired` is `DependenciesChanged`.